File: teamSkittles/atom-rest/src/main/java/com/example/atom/entities/Request.java
package com.example.atom.entities;

import com.example.atom.dto.ContractorDto;
import com.example.atom.dto.StateDto;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Request {
    @Id
    @Column(name = "id", nullable = false)
    private Long id;
    private String number; //номер заказа (по идее)
    private Date date;
    private String description;
    private Date releaseDate;


//    public RequestExtension(String name, String description) {
//        this.name = name;
//        this.description = description;
//    }
}

File: teamSkittles/atom-web/src/app/dto/Request.ts
import {Data} from "@angular/router";
import {State} from "./State";
import {Constractor} from "./Constractor";

export class Request {
  id: number;
  number: string;
  date: Data;
  contractor: Constractor;
  priority: number = 0;
  description: string;
  state: State;
  releaseDate: Data;

  constructor() {
  }
}
